Add Support for Typelib Packaging

danyeaw opened the pull request:
This PR adds support for installing girepository typelibs and adds additional default pkg-config locations.

Currently, the plugin is not packaging any typelibs. These allow for apps that use other language bindings to be packaged using linuxdeploy. For example, PyGObject uses the typelibs to introspect bindings for Python.
    
Additionally, this PR adds additional default locations for Ubuntu in case libgtk-3-dev and librsvg2-dev are not installed. A Python app wouldn't normally require these dependencies, so use default values as a fallback during packaging for linuxdeploy. Although I don't use Ubuntu myself, this is the typical setup, especially with CI runners and Docker.

Finally, it fixes a small issue with the symbolic link step at the end, which fails if the symbolic links already exist.

danyeaw committed:
Add support for installing girepository typelibs

This allows for apps that use other language bindings to be packaged
using linuxdeploy. For example, PyGObject uses the typelibs to
introspect bindings for Python.

danyeaw committed:
Add additional default pkg-config locations

Use default locations for Ubuntu in case libgtk-3-dev and librsvg2-dev are
not installed. A Python app wouldn't normally require these dependencies,
so use default values as a fallback during packaging for linuxdeploy.

danyeaw committed:
Fix failure if symbolic links already exist

TheTumultuousUnicornOfDarkness commented:
Hello,
I am skeptical about `/usr/lib/x86_64-linux-gnu`: it is too Debian/Ubuntu specific IMHO.
@TheAssassin what is your point of view about this kind of default?

Your PR is breaking CI pipelines: please fix.

TheAssassin commented:
I don't think these paths are a big problem They are used during build time only. Also, they are just fallback values, and additional libdirs could always be added, I guess. Typically, the value should be available from `pkg-config` anyway, right? It will work during runtime anyway, since we set the corresponding environment variables.

danyeaw committed:
Fix gobject-introspection-1.0 library not found

The default typelib directory was incorrectly pointing to a location
for rpm based distros which was causing CI to fail with a library
not found error.

danyeaw commented:
I 100% agree that the default paths are completely Debian/Ubuntu specific, but adding a fallback to a typical location I think only helps. I pushed a commit that fixes the default path for the typelibs which should fix the CI failure. :+1:

If we wanted to, we could remove libgtk-3-dev as a dependency for CI, because the default path should make it not needed.

danyeaw commented:
@X0rg Could you please restart the CI for me?

TheAssassin commented:
Done.

TheTumultuousUnicornOfDarkness commented:
Ok, if you are fine with these fallback paths, then it looks good to me.
@danyeaw please install missing dependencies in Dockerfiles.

danyeaw committed:
Add additional dependencies to test typelibs

danyeaw commented:
Dockerfiles updated!

danyeaw commented:
@X0rg or @TheAssassin could you please reapprove the checks? Thanks!

danyeaw commented:
Please approve the workflows again, thanks! :+1:

danyeaw commented:
One more time please, they are now all passing locally with Docker.

danyeaw committed:
Update dependencies for GTK4 to include typelibs

danyeaw commented:
I didn't realize this was going to impact the GTK4 builds as well, PR updated to include more dependencies for them. :+1: 

Could you please reapprove the workflow?

TheAssassin commented:
We get notifications for every commit. No need to double the amount of mails. You don't need to ask every time.

TheAssassin commented:
Also, you could test changes in your own repository, then you don't have to wait here.

danyeaw commented:
@TheAssassin Thanks, sorry for the doubling up. I have never run GitHub Actions on a fork of a repo, and it looks like I can enable them by going to the actions tab, thanks for the pointer!

It looks like all the checks finally passed, apologies again for having to reapprove the checks again and again, it was frustrating for me as well.

TheAssassin commented:
@X0rg maybe you could have another look? Please don't hesitate to merge if this works for you.

TheTumultuousUnicornOfDarkness commented:
LGTM.

danyeaw commented:
@X0rg and @TheAssassin thanks for your support helping to get this merged! 👍
